### DISCUSSION TITLE
Can we add NavigationView controls with Winui3 style and functionality?

### DISCUSSION THREAD

**Discussion by BoyFaceGirl** [dotnet/wpf#8198] [Q&A]
![image](https://github.com/dotnet/wpf/assets/8773178/2efd432a-de02-4ba2-a001-96baea7432d5)
Something like this?

**Comment by miloush** [comment]
Looks like `TreeView` or how does it differ?

**Reply by BoyFaceGirl** [reply]
Are you serious?If the treeview is transformed into a NavigationView in the WinUI style, it would be equivalent to redeveloping it. The treeview control is too weak, and whether it is the icon or the selected style, treeview is not suitable for navigation control at all. Otherwise, Microsoft will not redevelop the NavigationView control in the WinUI style

**Reply by miloush** [reply]
It was not clear from the screenshot what `NavigationView` is. From https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview I can see it is not just the tree view, it's adaptable with flyouts, has header and footer, back navigation, the suggestion box, settings. The tree view part supports separators and headers/grouping and invoking. That was my question.

Personally I think there are more simple and important controls that are missing - even the `NavigationView` itself relies on `AutoSuggestBox` for example - but I don't see any reason why navigation view cannot be added. PRs are welcome.

**Comment by jonasnordlund** [comment]
This one is now already part of WPF-UI that the refreshed .NET 9 theme is based on and a subset of, and I even found style resources for NavigationView in the .NET 9 WPF codebase, so I think this looks more like a "political" question about what and how much should be added to the WPF library. Right now, the line seems to have been drawn at "only augment existing controls and don't add new WinUI 3 ported controls". For those, you currently need to use the WPF-UI library. But with that one, you can absolutely have WinUI 3 style NavigationViews in WPF.

**Comment by NeverMorewd** [comment]
For your reference:
https://github.com/NeverMorewd/AsyncNavigation

**Reply by lindexi** [reply]
Awesome!
